lib/src/logic/questionnaire_controller.dart

```diff
@@ -85,72 +85,99 @@ class QuestionnaireController {
   }
 
   static Future<List<QuestionnaireItemBundle>> buildQuestionnaireItems(
-      Questionnaire questionnaire,
-      {Future<Attachment?> Function()? onAttachmentLoaded}) async {
+    List<QuestionnaireItem>? questionnaireItems, {
+    Future<Attachment?> Function()? onAttachmentLoaded,
+  }) async {
     List<QuestionnaireItemBundle> itemBundles = [];
+
```

Comment:
This is a breaking change, also the logic of this function is to generate the list of item bundles from a questionnaire, despite the fact that the function only uses the `questionnaire.item` property so far, the purpose is to have the full questionnaire instance to build the item bundles.

Reply:
I changed from `Questionnaire` to `List<QuestionnaireItem>` to be able to recursively call the method for the list of `QuestionnaireItem` inside QuestionnaireItems of type group. If someone is using `buildQuestionnaireItems` method out of `QuestionnaireController` class there will be a break in his code. but it is not huge break. so it is possible to introduce this change. 

if you wish to strictly keep the function signature like that. could you give me some suggestions on how you wish it to be?